### ISSUE TITLE
A few observersions:

### ISSUE THREAD

**Issue by tboege** [AlienWolfX/UZ801-USB_MODEM#10]
Thank you for your work - I will note af few observations. Hopefully some of them can be fixed:

1) The version of openwrt is rather old. /etc/banner:
`Based on Openwrt 21.02
 IOT Freedom For End-user!
 HandsomeMod 21.03-Attila-SNAPSHOT`
opkg points to  https://downloads.openwrt.org/releases/21.02-SNAPSHOT/targets/msm89xx... but that cannot be downloaded (not even if I replace https with http)
So It is not possible to get any extra packages
Will the required modifications be added to standard OpenWrt?

2) Only a part of 4GB flash is used:
`root@HandsomeMod:/tmp# df -h
Filesystem                Size      Used Available Use% Mounted on
/dev/root               126.3M    101.3M     22.4M  82% /
tmpfs                   192.6M    592.0K    192.0M   0% /tmp
tmpfs                   512.0K         0    512.0K   0% /dev`

The TX rate of wifi is limited to 6.5Mbs or 13.5Mbs - over ethernet I get >25Mbps download, and >10Mbps download

![image](https://github.com/user-attachments/assets/a0de551c-9fe0-49c0-bd63-dff590988a64)

3) In Wifi mode Client works ok, but if Client WDS it will nevet get ipadress.
3) In wifimode, the client will be disabled, if the default "SSID: HandsomeMod_A1377C | Mode: Master" is enabled.

![image](https://github.com/user-attachments/assets/ff7fa7b0-1070-447b-b763-2ae0f7b9ed92)

**Comment by AlienWolfX** [comment]
> Thank you for your work - I will note af few observations. Hopefully some of them can be fixed

Hello @tboege,

Thank you for your feedback. I'll review the issues that were identified and work on fixing them to the best of my ability.

Best regards, 
Allen

**Comment by AlienWolfX** [comment]
> Do you know if it is possible to enter edl or fastboot from your version of OpenWrt (or do I need to short D+ and GND). Maybee I can use reset button?

Yes, for fastboot execute `adb reboot-bootloader` and for edl `fastboot oem reboot-edl`

**Comment by tboege** [comment]
Thank you. I was not aware, that those options was available after
installation of openwrt

Den ons. 6. nov. 2024 kl. 17.29 skrev Allen Gabrielle Cruiz <
***@***.***>:

**Comment by alexandrglm** [comment]
> opkg points to https://downloads.openwrt.org/releases/21.02-SNAPSHOT/targets/msm89xx... but that cannot be downloaded (not even if I replace https with http) ...

OpenWRT doesn't officially support the msm8916 chipset, but some Git repositories offer opkg sources for it. These could be integrated into the build system or made available locally.

> In wifimode, the client will be disabled, if the default "SSID: HandsomeMod_A1377C | Mode: Master" is enabled.

As expected with OpenWRT, enabling the default Wi-Fi access point ('SSID: HandsomeMod_A1377C | Mode: Master') disables the client interface.  When radio0 is configured as an AP, the client configuration is disabled, and viceversa.

Hope this helps.

Alexandr.

**Comment by samthelion17** [comment]
> but some Git repositories offer opkg sources for it

Where can we find some of those repositories? I just want to install wireguard.

Thanks for all the work you've put into this project! :heart:

**Comment by AlienWolfX** [comment]
Hello @samthelion17,

> How can I query the supported bands and other info like imei with mmcli as you wrote in the README?

You can run `mmcli -m 0` to query modem information

> Where can we find some of those repositories? I just want to install wireguard.

> [!NOTE]
I haven't full tested this but [Here's](https://www.mediafire.com/file/gjlysk0ea6nvswc/uz801_with_wireguard.rar/file) a compiled version with wireguard

**Comment by samthelion17** [comment]
> I haven't full tested this but [Here's](https://www.mediafire.com/file/gjlysk0ea6nvswc/uz801_with_wireguard.rar/file) a compiled version with wireguard

Can confirm that wireguard is working, and it's working fast! I had to reboot (unplug, re-plug?) after creating and configuring the interface but other than that it works great.

**Comment by AlienWolfX** [comment]
> Can confirm that wireguard is working, and it's working fast! 

Thank you for your feedback. ♥️

Best regards,
Allen.

**Comment by alexandrglm** [comment]
You may restart with /etc/init.d/the_service restart:
e.g.
# /etc/init.d/network restart

**Comment by AlienWolfX** [comment]
* Whole flash is used
* TX Rate Fixed

Build: [OpenWRT-UZ801_2026](https://github.com/AlienWolfX/HandsomeMod-UZ801/releases/download/v1.0-alpha/OpenWRT-UZ801_2026.tar.xz)
